compute_transfer_examples/collection_transfer_requires_flow/compute_transfer_example_1_definition.json

```diff
@@ -0,0 +1,122 @@
+{
+  "StartAt": "SetConstants",
+  "States": {
+    "SetConstants": {
+      "Type": "ExpressionEval",
+      "Parameters": {
+        "gcs_base_path": "/",
+        "compute_endpoint_id": "<INSERT YOUR COMPUTE ENDPOINT ID HERE>",
+        "compute_function_id": "<INSERT YOUR COMPUTE FUNCTION ID HERE>",
+        "compute_output_directory.=": "'/' + _context.run_id + '/'"
+      },
+      "ResultPath": "$.constants",
+      "Next": "GetDestinationPath"
+    },
+    "GetDestinationPath": {
+      "Type": "ExpressionEval",
+      "Parameters": {
+        "path.=": "getattr('destination_path', transfer_settings.DATA[0].destination_path)"
+      },
+      "ResultPath": "$.destination_path",
+      "Next": "InitIterateSourcePaths"
+    },
+    "InitIterateSourcePaths": {
+      "Type": "ExpressionEval",
+      "Parameters": {
+        "src_paths_iterator_pos": 0,
+        "src_paths": []
+      },
+      "ResultPath": "$.iterate_source_paths",
+      "Next": "IterateSourcePathsConditional"
+    },
+    "IterateSourcePathsConditional": {
+      "Type": "ExpressionEval",
+      "Parameters": {
+        "conditional.=": "len(transfer_settings.DATA) > iterate_source_paths.src_paths_iterator_pos"
+      },
+      "ResultPath": "$.iterate_source_paths_conditional",
+      "Next": "IterateSourcePaths"
+    },
+    "IterateSourcePaths": {
+      "Type": "Choice",
+      "Choices": [
+        {
+          "Variable": "$.iterate_source_paths_conditional.conditional",
+          "BooleanEquals": true,
+          "Next": "AppendSourcePath"
+        }
+      ],
+      "Default": "MakeComputeWorkingDir"
+    },
+    "AppendSourcePath": {
+      "Type": "ExpressionEval",
+      "Parameters": {
+        "src_paths.=": "iterate_source_paths.src_paths + [transfer_settings.DATA[iterate_source_paths.src_paths_iterator_pos].source_path]",
+        "src_paths_iterator_pos.=": "iterate_source_paths.src_paths_iterator_pos + 1"
+      },
+      "ResultPath": "$.iterate_source_paths",
+      "Next": "IterateSourcePathsConditional"
+    },
```

Comment:
I think this entire iteration over `source_path` values can be expressed in a single `ExpressionEval`, rather than a series of states which create a loop.
Something like (untested) this:
```json
    "CollectSourcePaths": {
      "Type": "ExpressionEval",
      "Parameters": {
        "src_paths.=": "[item.source_path for item in transfer_settings.DATA]",
      },
    },
```